src/Hazelcast.Net.Tests/Clustering/ConnectMembersTests.cs

```diff
@@ -68,53 +70,66 @@ await foreach (var (member, token) in memberConnectionQueue.WithCancellation(can
                 Assert.That(addresses, Does.Contain(NetworkAddress.Parse("127.0.0.1:2")));
             }, 2000, 200);
 
+
             // -- can suspend while waiting
 
-            queue.Suspend(); // suspend
+            await queue.SuspendAsync(); // suspend
 
             queue.Add(MemberInfo(NetworkAddress.Parse("127.0.0.1:3")));
 
             await Task.Delay(500);
             Assert.That(addresses.Count, Is.EqualTo(2)); // nothing happened
 
-            queue.Resume(); // resume => will process
+            queue.Resume(); // resume => processes the queue
 
             await AssertEx.SucceedsEventually(() =>
             {
                 Assert.That(addresses.Count, Is.EqualTo(3));
                 Assert.That(addresses, Does.Contain(NetworkAddress.Parse("127.0.0.1:3")));
             }, 2000, 200);
 
-            // -- can suspend while connecting
 
-            await mutex.WaitAsync(); // block
+            // -- suspending waits for the current connection
+
+            await mutex.WaitAsync(); // blocks the connections
 
             queue.Add(MemberInfo(NetworkAddress.Parse("127.0.0.1:4")));
             await Task.Delay(500);
 
-            queue.Suspend();
-            mutex.Release(); // resume => should cancel current connect
+            var task = queue.SuspendAsync();
+            Assert.That(task.IsCompleted, Is.False);
```

Comment:
Is there any specific reason to use always `Assert.That`
What about `Assert.IsTrue(task.IsCompleated)` for true false checks